src/classic/class/ReactClass.js

```diff
@@ -828,6 +828,12 @@ var ReactClass = {
       // This constructor is overridden by mocks. The argument is used
       // by mocks to assert on what gets mounted.
 
+      invariant(
+        this,
```

Comment:
Yea, you're right. Also I'm going to make it a warning in __DEV__ only so we aren't doing instanceof all the time.